packages/docusaurus/src/webpack/utils.ts

```diff
@@ -143,11 +144,16 @@ export function getBabelOptions({
 
 export function getBabelLoader(
   isServer: boolean,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   babelOptions?: TransformOptions | string,
 ): Loader {
   return {
-    loader: require.resolve('babel-loader'),
-    options: getBabelOptions({isServer, babelOptions}),
+    loader: require.resolve('esbuild-loader'),
+    options: {
+      loader: 'tsx',
+      format: isServer ? 'cjs' : 'esm',
+      target: 'es2017',
+    },
   };
 }
```

Comment:
I plan to make this function be like

```js
if (process.env.ESBUILD) {
  return {
    loader: require.resolve('esbuild-loader'),
    options: {
      loader: 'tsx',
      format: isServer ? 'cjs' : 'esm',
      target: 'es2017',
    },
  };
  return {
    loader: require.resolve('babel-loader'),
    options: getBabelOptions({isServer, babelOptions}),
  }
}
```

In this way all the existing code that uses the `getBabelLoader` can work without modification. This is somehow needed since this function is exposed [here](https://docusaurus.io/docs/lifecycle-apis#utils). However, this is bad since the function should then really be named as `getJSLoader`. Or we can create a new function called `getJSLoader` and deprecate the `getBabelLoader` function.

Do maintainers have any options?

Reply:
@lex111 @slorber

Reply:
`getJSLoader` is also something I had in mind. Would be fine to deprecate getBabelLoader()